ValkeyVectorSearch.md

```diff
@@ -0,0 +1,154 @@
+---
+RFC: <PR number>
+Status: <Proposed>
+---
+
+# Valkey Vector Search RFC
+
+## Abstract
+
+This proposed Valkey Vector Search module will introduce KNN (“Flat”) and ANN (“HNSW”) vector similarity search algorithms to provide vector search capabilities on both Valkey Cluster and Valkey Standalone. With vector search on Valkey, users can take advantage of low latency in-memory vector search in their familiar Valkey environment. 
+
+In its current status, this RFC is meant as a placeholder and is not ready for review by the Valkey TSC yet. Google Cloud and AWS both have vector search implementations with different design specs, 
+so the design section is not fully complete yet. We will publish performance benchmarks once complete and adjust the design section to ask the TSC for feedback on both of the vector search implementations on Valkey.
+
+The original RFC is a collaborative effort between Google Cloud (Kyle Meggs, Yair Gottdenker) and AWS (Sanjit Misra, Allen Samuels). We welcome comments and suggestions to improve the RFC.
+
+## Motivation
+
+With the rising popularity of generative AI, developers have turned to in-memory data stores such as Valkey or Redis for ultra-low latency (p99 single digit ms) vector search for use cases like Retrieval Augmented Generation (RAG), recommendation systems, semantic search, semantic caching, and more. For developers already using Valkey or Redis, co-locating their vectors and search functionality alongside their underlying data will accelerate and simplify adoption. And for other “greenfield” users, the ultra-low latency of Valkey vector search may attract them to replace their existing vector database for high performance use cases to achieve in-memory speeds at the highest levels of recall and throughput.
+
+## Design considerations
+
+Here we list Critical User Journeys (CUJs) which we believe Valkey users will desire as part of a Vector Search Module. 
+
+<br>
+
```

Comment:
Please delete these `<br>`s (not only this one). They makes the markdown source code harder to read.

If the spacing between paragraphs looks too small, that depends on the CSS, but we can live with that.

```suggestion

```